ibis/common/patterns.py

```diff
@@ -797,6 +823,8 @@ class InstanceOf(Slotted, Singleton, Pattern):
     """
 
     __slots__ = ("type",)
+    # Used Any as not sure how to type this
+    type: Any
```

Comment:
This is directly passed to isinsance as typeinfo. Probably doesn't worth the trouble since it can be: `type`, `tuple[type, ...]`, `UnionTypeType`

Reply:
I now annotated it as `_ClassInfo` which I defined as:

```python
# Keep this alias in sync with unittest.case._ClassInfo
if sys.version_info >= (3, 10):
    _ClassInfo: TypeAlias = type | UnionType | tuple["_ClassInfo", ...]
else:
    _ClassInfo: TypeAlias = type | tuple[_ClassInfo, ...]
```

This is what is used in the [standard library typeshed](https://github.com/python/typeshed/blob/main/stdlib/builtins.pyi#L1467) for `isinstance`. What do you think?